coordinatorlib/src/main/resources/com.socrata.datacoordinator.truth.schema/20171122-add-secondary-jobs-tables.xml

```diff
@@ -0,0 +1,75 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+    http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-2.0.xsd">
+    <changeSet author="Alexa Rust" id="20171122-add-secondary-jobs-tables">
+        <preConditions onFail="MARK_RAN">
+            <and>
+                <not>
+                    <tableExists tableName="secondary_add_jobs"/>
+                </not>
+                <not>
+                    <tableExists tableName="secondary_delete_jobs"/>
+                </not>
+            </and>
+        </preConditions>
+        <createTable tableName="secondary_add_jobs">
+            <column name="dataset_system_id" type="BIGINT">
+                <constraints nullable="false"
+                             foreignKeyName="secondary_add_jobs_dataset_system_id_fkey"
+                             references="dataset_map(system_id)"
+                />
+            </column>
+            <column name="store_id" type="VARCHAR(64)">
+                <constraints nullable="false"/>
+            </column>
+            <column name="job_id" type="UUID">
+                <constraints nullable="false"/>
+            </column>
+            <column name="step" type="SMALLINT">
+                <constraints nullable="false"/>
+            </column>
+            <column name="created_at" type="TIMESTAMP WITH TIME ZONE" defaultValueComputed="now()">
+                <constraints nullable="false"/>
+            </column>
+        </createTable>
+        <createIndex tableName="secondary_add_jobs" indexName="secondary_add_jobs_dataset_system_id_store_id">
+            <column name="dataset_system_id"/>
+            <column name="store_id"/>
+        </createIndex>
+        <createIndex tableName="secondary_add_jobs" indexName="secondary_add_jobs_job_id_step" unique="true">
+            <column name="job_id"/>
+            <column name="step"/>
+        </createIndex>
+
+        <createTable tableName="secondary_delete_jobs">
```

Comment:
Have we thought about using one job table for both add and delete?  Or even one job row to represent both add and delete?  That seems easier to support chronological ordering in case there is a need.

Reply:
We spent some time talking about both those options. Decided against representing an add and a delete together in a single row because we don't want to have to update. This seemed like the simplest option. When data-coordinator makes a decision about how to move things, it can right a delete and add jobs to respective tables. Then the secondary-watcher depending on what it is doing can look for those jobs and delete them once complete.